obspy/core/tests/test_dict_conversion.py

```diff
@@ -0,0 +1,43 @@
+"""
+Testing utilities for ObsPy.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import unittest
+
+import obspy
+from obspy.core.event.dict import catalog_to_dict, dict_to_catalog
+from obspy.core.util.testing import create_diverse_catalog
+
+TEST_CATALOGS = [obspy.read_events(), create_diverse_catalog()]
```

Comment:
I wonder if there is a simple way to get all the catalogs that the different `io` modules use for their tests and test them?

Reply:
I am not sure if there is a simple way but it wouldn't be too hard to manually specify their test data paths. Might be worth adding something to the core.util.testing module for easily collecting all of them.